agent/xdsv2/testdata/input/l4-single-implicit-destination-tproxy.golden

```diff
@@ -0,0 +1,85 @@
+{
+  "proxyState":  {
+    "identity":  {
+      "tenancy":  {
+        "partition":  "default",
+        "namespace":  "default",
+        "peerName":  "local"
+      },
+      "name":  "test-identity"
+    },
+    "listeners":  [
+      {
+        "name":  "outbound_listener",
+        "direction":  "DIRECTION_OUTBOUND",
+        "hostPort":  {
+          "host":  "127.0.0.1",
+          "port":  15001
+        },
+        "routers":  [
+          {
+            "match":  {
+              "prefixRanges":  [
+                {
+                  "addressPrefix":  "1.1.1.1",
+                  "prefixLen":  32
+                }
+              ],
+              "destinationPort": 8080
+            },
+            "l4":  {
+              "name":  "tcp.api-1.default.dc1.internal.foo.consul",
+              "statPrefix":  "upstream.tcp.api-1.default.default.dc1"
+            }
+          }
+        ],
+        "capabilities":  [
+          "CAPABILITY_TRANSPARENT"
+        ]
+      }
+    ],
+    "clusters":  {
+      "tcp.api-1.default.dc1.internal.foo.consul":  {
+        "endpointGroup":  {
+          "dynamic":  {
+            "config":  {
+              "disablePanicThreshold":  true
+            },
+            "outboundTls":  {
+              "outboundMesh":  {
+                "identityKey":  "test-identity",
+                "validationContext":  {
+                  "spiffeIds":  [
+                    "spiffe://foo.consul/ap/default/ns/default/identity/api1-identity"
+                  ]
+                },
+                "sni":  "api-1.default.dc1.internal.foo.consul"
+              },
+              "alpnProtocols": [
+                "consul~tcp"
+              ]
+            }
+          }
+        }
+      }
+    }
+  },
+  "requiredEndpoints":  {
+    "api-1.default.dc1.internal.foo.consul":  {
+      "id":  {
+        "name":  "api-1",
+        "type":  {
+          "group":  "catalog",
+          "groupVersion":  "v1alpha1",
+          "kind":  "ServiceEndpoints"
+        },
+        "tenancy":  {
+          "partition":  "default",
+          "namespace":  "default",
+          "peerName":  "local"
+        }
+      },
+      "port":  "mesh"
+    }
+  }
+}
```

Comment:
Is this a copy-paste of the existing tests? If so, how were you thinking we'd keep these from diverging?

Reply:
The other tests are golden files of proxyState, testing .  This is a golden files for catalog/mesh vs -> proxy State.  

This test and file is for testing proxyState -> xds conversion.  We have xds golden file tests that test proxyState -> xds, but they are proxy state converter tests, so adding those there did not seem appropriate.  I added there was nothing testing that an ALPN router actually got converted to XDS.  No tests failed and it was not until running an integration test that I noticed this it was even missing.  So I added this to ensure that it happened.  This type of testing will continue to be needed as we expand the types of Matches we support as we add more functionality.